### PR TITLE
[serial] Allow custom baudrate and add option 2400

### LINKED ISSUE #13131: [serial] Support custom baudrates
Hey Team,
In OH3.2.0 custom Baud rates worked
upgraded to OH3.3.0 and custom Baud rates no longer work

[Serial binding other baud rates - Community](https://community.openhab.org/t/serial-binding-other-baud-rates/137273)

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #13131

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by jlaur** [comment]
@lolodomo - broken in 3.3 as consequence of #2720?

**Comment by lolodomo** [comment]
This PR adds the ability to use 2400 or any custom baudrate. 
No link with core PR.

**Comment by jlaur** [comment]
@lolodomo - yes, was just looking for confirmation that this was a regression caused by mentioned PR in core. :)

**Review by jlaur — APPROVED** [review]
Thanks for the fix!

**Comment by lolodomo** [comment]
In fact, the issue is mentioning 2 issues, the one with decimal value which is a core problem, and the one I am fixing with this PR.

**Comment by jlaur** [comment]
@lolodomo - okay, I'm assuming this one was indeed a regression in 3.3 caused by a change in core, so I have switched to "Bug" as label.

File: bundles/org.openhab.binding.serial/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -26,8 +26,10 @@
 				<advanced>true</advanced>
 				<label>Baud Rate</label>
 				<description>Set the baud rate</description>
+				<limitToOptions>false</limitToOptions>
```
Comment (MikeJMajor) [inline bundles/org.openhab.binding.serial/src/main/resources/OH-INF/thing/thing-types.xml:29]: When I tried adding this it stops the UI displaying the options and just presents a box to type in any number. What do you see?

**Review by MikeJMajor — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.serial/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.serial/src/main/resources/OH-INF/thing/thing-types.xml:29]: Yes, apparently in that case, Main UI does not show the options. But at least, you can enter any baudrate.

**Pull request merged**